src/main/kotlin/com/salesforce/zero/quickquartz/ExposedTriggers.kt

```diff
@@ -0,0 +1,206 @@
+            startTime, endTime, calendarName, misfireInstr, jobData)
+    }
+}
+
+/**
+ * This is a lambda whose receiver (this) is a BatchInsertStatement.
+ * It has one param (a TriggerEntity) and it returns Unit.
+ * <p/>
+ * Usage:
+ * ```
+ * transaction {
+ *      QuickQuartzTriggers.batchInsert(data = triggers, body = batchInsertTriggers)
+ * }
+ * ```
+ */
+val batchInsertTriggers: BatchInsertStatement.(TriggerEntity) -> Unit = { trigger ->
+    this[schedName] = trigger.schedName
+    this[triggerName] = trigger.triggerName
+    this[triggerGroup] = trigger.triggerGroup
+    this[jobName] = trigger.jobName
+    this[jobGroup] = trigger.jobGroup
+    this[description] = trigger.description
+    this[nextFireTime] = trigger.nextFireTime
+    this[prevFireTime] = trigger.prevFireTime
+    this[priority] = trigger.priority
+    this[triggerState] = trigger.triggerState
+    this[triggerType] = trigger.triggerType
+    this[startTime] = trigger.startTime
+    this[endTime] = trigger.endTime
+    this[calendarName] = trigger.calendarName
+    this[misfireInstr] = trigger.misfireInstr
+    this[jobData] = trigger.jobData
+}
+
+/**
+ * helper that reads from a result row to a TriggerEntity
+ *
+ * Usage:
+ * ```
+ * (QuickQuartzJobDetails innerJoin QuickQuartzTriggers).selectAll().map { it.toTriggers() }
+ * ```
+ */
+fun ResultRow.toTrigger(): TriggerEntity = TriggerEntity(
+    schedName = this[schedName],
+    triggerName = this[triggerName],
+    triggerGroup = this[triggerGroup],
+    jobName = this[jobName],
+    jobGroup = this[jobGroup],
+    description = this[description],
+    nextFireTime = this[nextFireTime],
+    prevFireTime = this[prevFireTime],
+    priority = this[priority],
+    triggerState = this[triggerState],
+    triggerType = this[triggerType],
+    startTime = this[startTime],
+    endTime = this[endTime],
+    calendarName = this[calendarName],
+    misfireInstr = this[misfireInstr],
+    jobData = this[jobData]
+)
+
+/**
+ * maps a quartz Trigger to TriggerEntity
+ */
+val toQuickQuartzTrigger: Trigger.() -> TriggerEntity = {
+    TriggerEntity(
+        triggerName = this.key.name,
+        triggerGroup = this.key.group,
+        jobName = this.jobKey.name,
+        jobGroup = this.jobKey.group,
+        description = this.description,
+        nextFireTime = this.nextFireTime?.time,
+        prevFireTime = this.previousFireTime?.time,
+        priority = this.priority,
+        // triggerState = this.     // FIXME how do we get the state?
+        // triggerType =            // FIXME
+        startTime = this.startTime.time,
+        endTime = this.endTime.time,
+        calendarName = this.calendarName,
+        misfireInstr = this.misfireInstruction,
+        /**
+         * like job detail, [org.quartz.impl.triggers.AbstractTrigger.getJobDataMap] has the same problem
+         */
+        jobData = QuickSerializer.serialize(this.jobDataMap.takeUnless { it.isEmpty() })
```

Comment:
Do we really need a full fledged, non-threadsafe, serialization library when all we're "serializing" is maps of String->String?

For example, Kryo by default theoretically requires class registration for every class you will serialize/deserialize (https://github.com/EsotericSoftware/kryo#registration), and if you screw up the ordering (or change it), things will fail. I don't see that happening here, so maybe you don't have to declare Map<String,String> because they're base types, but then isn't the fact that QuickSerializer just exposes T a risk?

Another example, if we ever have to upgrade it (https://github.com/EsotericSoftware/kryo#kryo-versioning-and-upgrading), even in a minor version upgrade "some minor breakage is allowed", which sounds fun for something we don't need?

Reply:
Hm, so I would prefer to stick to a binary format, but I'm not married to Kryo at all. We could switch from a byte array to a JSONB (which postgres happily supports) and just go with that? 🤷‍♂